### PR TITLE
[Issue 13854][broker] Fix call sync method in async rest api for internalDeleteSubscription #13884

### PR DESCRIPTION
**Pull request by codertmy**
Master Issue: #13854

### Motivation
See #13854

### Verifying this change
- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:
  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation
Check the box below or label this PR directly (if you have committer privilege).
Need to update docs? 
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -1556,66 +1558,67 @@ private void internalDeleteSubscriptionForNonPartitionedTopic(AsyncResponse asyn
...
+                        internalDeleteSubscriptionForNonPartitionedTopicForcefully(asyncResponse, subName, authoritative);
+                    }
+                }).exceptionally(ex -> {
+                    log.error("[{}] Failed to delete subscription forcefully {} from topic {}", clientAppId(), subName, topicName, ex);
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
+                    return null;
+                });
+            }
+        }).exceptionally(e -> {
+            log.error("[{}] Failed to delete subscription {} from topic {}", clientAppId(), subName, topicName, e);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1618]: Need check getCause

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -1556,66 +1558,67 @@ private void internalDeleteSubscriptionForNonPartitionedTopic(AsyncResponse asyn
...
+                            }
+
+                            asyncResponse.resume(Response.noContent().build());
+                            return null;
+                        });
+                    } else {
+                        internalDeleteSubscriptionForNonPartitionedTopicForcefully(asyncResponse, subName, authoritative);
+                    }
+                }).exceptionally(ex -> {
+                    log.error("[{}] Failed to delete subscription forcefully {} from topic {}", clientAppId(), subName, topicName, ex);
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:1612]: Code style

**Review by Jason918 — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 2 above)
Comment (codertmy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Ok, I will fix it

**Review by Jason918 — APPROVED** [review]

**Comment by Jason918** [comment]
@codelipenghui @mattisonchao @liudezhi2098 @gaozhangmin PTAL

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**